pkg/kubectl/cmd/apply.go

```diff
@@ -240,6 +240,17 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, opti
 			visitedNamespaces.Insert(info.Namespace)
 		}
 
+		// Add change-cause annotation to resource info if it should be recorded
+		if cmdutil.ShouldRecord(cmd, info) {
+			recordInObj := info.VersionedObject
```

Comment:
Actually, both `info.Object` and `info.VersionedObject` are updated with the annotation (otherwise the test-cmd would have failed). It seems that changes made to meta.Accessor affect both objects.